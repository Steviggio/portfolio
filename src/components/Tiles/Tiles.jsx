import React from "react";

// Not currently used in the portfolio
const Tiles = ({datas}) => {

  return (
    <>
      <div className="flex flex-col w-2/6 gap-6 items-center p-5 bg-light-blue rounded-md shadow-md h-fit">
        <img className="h-fit w-full  rounded-xl shadow-sm" 
        src="" 
        alt="" />
        <div className='h-fit w-fit flex flex-col justify-center pr-4 bg-white rounded-lg  shadow-md'>
          <h5 className='underline text-xl'>Langages utilisÃ©s</h5>
          <ul className='flex gap-10 justify-center m-3'>{datas.map((item) => {
            return (
              <li key={item.key}><img className='w-20' src={item.iconUrl} alt={`${item.name} logo`} /></li>
            )
          })}</ul>
        </div>
      </div>
    </>
  )
}

export default Tiles;