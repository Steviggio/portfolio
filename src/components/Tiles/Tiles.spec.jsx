import React from 'react';
import renderer from 'react-test-renderer';
import Tiles from './Tiles';

// Mock data for testing
const mockData = [
  {
    key: 1,
    iconUrl: 'icon1.png',
    name: 'Language 1',
  },
  {
    key: 2,
    iconUrl: 'icon2.png',
    name: 'Language 2',
  },
];

describe('Tiles Component', () => {
  it('renders the component with mock data', () => {
    const component = renderer.create(<Tiles datas={mockData} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
