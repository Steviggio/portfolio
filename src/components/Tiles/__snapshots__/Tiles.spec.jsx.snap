// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tiles Component renders the component with mock data 1`] = `
<div
  className="flex flex-col w-2/6 gap-6 items-center p-5 bg-light-blue rounded-md shadow-md h-fit"
>
  <img
    alt=""
    className="h-fit w-full  rounded-xl shadow-sm"
    src=""
  />
  <div
    className="h-fit w-fit flex flex-col justify-center pr-4 bg-white rounded-lg  shadow-md"
  >
    <h5
      className="underline text-xl"
    >
      Langages utilis√©s
    </h5>
    <ul
      className="flex gap-10 justify-center m-3"
    >
      <li>
        <img
          alt="Language 1 logo"
          className="w-20"
          src="icon1.png"
        />
      </li>
      <li>
        <img
          alt="Language 2 logo"
          className="w-20"
          src="icon2.png"
        />
      </li>
    </ul>
  </div>
</div>
`;
